src/sagas/projects.js

```diff
@@ -47,6 +50,19 @@ export function* changeCurrentProject() {
   yield call(saveCurrentProject, state);
 }
 
+export function* rehydrateProject({payload: {rehydratedProject}}) {
+  const state = yield select();
+  const allProjectKeys = getProjectKeys(state);
+  if (allProjectKeys.includes(rehydratedProject.projectKey)) {
+    yield put(__changeCurrentProject(rehydratedProject.projectKey));
+  } else {
+    rehydratedProject.projectKey = generateProjectKey();
+    rehydratedProject.updatedAt = null;
```

Comment:
@ajgreenb OK so I just checked this out and poked around with it. A couple of thoughts:

1. This is really cool and a total game changer.
2. If you simply remove the above two lines, as well as the entire `if` branch, you should be good. By simply keeping the `projectKey` stable throughout the process, key collisions ensure that we never get a duplicate when we don’t want one.

I have some more nuts-and-bolts feedback on the code as well but I’ll hold off until you’ve got it where you want it. Let me know!